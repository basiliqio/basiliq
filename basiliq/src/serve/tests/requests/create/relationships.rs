use super::*;

#[basiliq_test(run_migrations)]
async fn single_rel(pool: sqlx::PgPool) {
    let state = prepare_basiliq_test(pool).await;
    let request = prepare_basiliq_request(
        Method::POST,
        "/public__favorite_color",
        Body::from(
            json!({
                "data": json!({
                    "type": "public__favorite_color",
                    "attributes": json!({
                        "color": "rot"
                    })
                })
            })
            .to_string(),
        ),
    );

    let resp = crate::serve::main_service(state.clone(), request)
        .await
        .unwrap();

    assert_eq!(resp.status(), StatusCode::CREATED);
    let res = handle_create(resp).await;
    crate::test_json!(res);
    let color_id = res.as_object().and_then(|x| x.get("id")).unwrap().clone();
    let request = prepare_basiliq_request(
        Method::POST,
        "/public__peoples",
        Body::from(
            json!({
                "data": json!({
                    "type": "public__peoples",
                    "attributes": json!({
                        "first-name": "AAAAA",
                        "last-name": "BBBBBBB"
                    }),
                    "relationships": json!({
                        "public__favorite_color": json!({
                            "data": json!({
                                "id": color_id,
                                "type": "public__favorite_color"
                            })
                        })
                    })
                })
            })
            .to_string(),
        ),
    );
    let resp = crate::serve::main_service(state.clone(), request)
        .await
        .unwrap();
    println!("{:#?}", resp);
    let peoples = state
        .store()
        .ciboulette()
        .get_type("public__peoples")
        .unwrap();
    for rel in peoples.relationships() {
        println!("{}", rel.0);
    }
    assert_eq!(resp.status(), StatusCode::CREATED);
    let res = handle_create(resp).await;
    crate::test_json!(res);
}
