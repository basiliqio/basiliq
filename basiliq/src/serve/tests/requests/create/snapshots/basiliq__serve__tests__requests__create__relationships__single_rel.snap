---
source: basiliq/src/serve/tests/requests/create/relationships.rs
expression: res

---
{
  "type": "public__favorite_color",
  "id": "[uuid]",
  "attributes": {
    "color": "rot"
  }
}
